legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/tests/testSpannerCompatibility.pure

```diff
@@ -0,0 +1,351 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+###Pure
+Class meta::relational::spanner::tests::domain::Person
+{
+   firstName: String[1];
+   lastName: String[1];
+   age:Integer[1];
+   key:Integer[1];
+}
+
+Class meta::relational::spanner::tests::domain::Thing
+{
+   name: String[1];
+   id: Integer[1];
+}
+
+###Relational
+Database meta::relational::spanner::tests::db::spannerDb
+(
+   Table "persontable" (ID INT PRIMARY KEY, firstName VARCHAR(200), lastName VARCHAR(200), age INT, key INT)
+   Schema database1
+   (
+      Table "thingtable" (ID INT PRIMARY KEY, NAME VARCHAR(200))
+   ) 
+   
+   Join Person_Person("persontable".key = {target}.key)
+)
+
+###Mapping
+Mapping meta::relational::spanner::tests::mapping::simpleRelationalMappingPersonForSpanner
+(
+   meta::relational::spanner::tests::domain::Person : Relational
+   {
+      scope([meta::relational::spanner::tests::db::spannerDb])
+      (
+         firstName : "persontable".firstName,
+         lastName: "persontable".lastName,
+         age : @Person_Person|"persontable".key         
+      )
+   }
+
+   meta::relational::spanner::tests::domain::Thing : Relational
+   {
+    ~primaryKey
+    (
+      [meta::relational::spanner::tests::db::spannerDb]database1."thingtable".ID
+    )
+    ~mainTable [meta::relational::spanner::tests::db::spannerDb]database1."thingtable"
+    id: [meta::relational::spanner::tests::db::spannerDb]database1."thingtable".ID,
+    name: [meta::relational::spanner::tests::db::spannerDb]database1."thingtable".NAME
+  }
+      
+)
+
+###Pure
+import meta::relational::runtime::*;
+import meta::relational::metamodel::execute::*;
+import meta::relational::mapping::*;
+import meta::relational::functions::asserts::*;
+import meta::relational::runtime::*;
+import meta::relational::tests::model::simple::*;
+import meta::relational::tests::*;
+import meta::relational::functions::sqlstring::*;
+import meta::relational::tests::mapping::sqlFunction::model::domain::*;
+import meta::relational::tests::mapping::sqlFunction::model::mapping::*;
+import meta::pure::runtime::*;
+import meta::pure::profiles::*;
+
+function <<test.BeforePackage>> meta::relational::spanner::tests::setUp():Boolean[1]
```

Comment:
The package path here seems a bit off !!

Reply:
Ok. But what should it be instead ?  For example meta::pure::router tests are in meta::relational::tests::routing.

Reply:
How about meta::relational::sqlQueryToString::tests::spanner ?
Because we are essentially testing sqlQueryToString code